File: backend/app/schemas/customer.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class CustomerCreate(BaseModel):
    user_id: uuid.UUID
    external_id: str
    email: str
    company_name: str | None = None
    contact_name: str | None = None
    phone: str | None = None
    custom_fields: dict[str, object] = Field(default_factory=dict)


class CustomerRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    external_id: str
    email: str
    company_name: str | None
    contact_name: str | None
    phone: str | None
    custom_fields: dict[str, object]
    created_at: datetime
    updated_at: datetime


class CustomerUpdate(BaseModel):
    external_id: str | None = None
    email: str | None = None
    company_name: str | None = None
    contact_name: str | None = None
    phone: str | None = None
    custom_fields: dict[str, object] | None = None

File: frontend/src/features/customers/types/customer.ts
export type Customer = {
  id: string
  user_id: string
  external_id: string
  email: string
  company_name: string | null
  contact_name: string | null
  phone: string | null
  custom_fields: Record<string, unknown>
  created_at: string
  updated_at: string
}

export type CustomersPagination = {
  page: number
  page_size: number
  total: number
  total_pages: number
}

export type CustomersListResult = {
  data: Customer[]
  pagination: CustomersPagination
}
